src/main/java/com/foodybuddy/service/impl/OrderDishServiceImpl.java

```diff
@@ -0,0 +1,155 @@
+package com.foodybuddy.service.impl;
+
+import java.util.List;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.springframework.stereotype.Service;
+
+import com.foodybuddy.dao.DishDAO;
+import com.foodybuddy.dao.OrderDishDAO;
+import com.foodybuddy.dao.impl.DishDAOImpl;
+import com.foodybuddy.dao.impl.OrderDishDAOImpl;
+import com.foodybuddy.model.Dish;
+import com.foodybuddy.model.Order;
+import com.foodybuddy.model.OrderDish;
+import com.foodybuddy.service.OrderDishService;
+import com.foodybuddy.utils.SessionFactoryUtils;
+
+@Service
+public class OrderDishServiceImpl implements OrderDishService {
+	private SessionFactory sessionFactory;
+
+	public List<OrderDish> getListOrderByDishId(int dishId) throws Exception {
+		sessionFactory = SessionFactoryUtils.getSessionFactory();
+		Session session = null;
+		try {
+			if (dishId <= 0) {
+				throw new Exception("invalid dishId ");
+			}
+			session = this.sessionFactory.openSession();
+			OrderDishDAO orderDishDAO = new OrderDishDAOImpl(session);
+			List<OrderDish> orderDishList = orderDishDAO.getListByDishId(dishId);
+			return orderDishList;
+		} catch (Exception ex) {
+			throw new Exception("ListingbyDishid failed: " + ex.getMessage(), ex);
+		} finally {
+			if (session != null) {
+				session.close();
+			}
+		}
+	}
+
+	public OrderDishServiceImpl(SessionFactory sessionFactory) {
+		this.sessionFactory = sessionFactory;
+	}
+
+	public List<OrderDish> getListOrderByOrderId(int orderId) throws Exception {
+		sessionFactory = SessionFactoryUtils.getSessionFactory();
```

Comment:
Why again fetching sessionFactory ? We have the global session factory which we get from constructor